components/Tabs/examples/MDCTabBarViewTypicalExampleViewController.m

```diff
@@ -265,6 +267,47 @@ - (void)changeItemsToTextAndImage {
   }
 }
 
+#pragma mark - UIScrollViewDelegate
+
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
+  [self logVisibleItems];
+}
+
+#pragma mark - UIViewController
+
+- (void)viewWillTransitionToSize:(CGSize)size
+       withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
+  [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+  [coordinator
+      animateAlongsideTransition:nil
+                      completion:^(
+                          id<UIViewControllerTransitionCoordinatorContext> _Nonnull context) {
+                        [self logVisibleItems];
+                      }];
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+  [self logVisibleItems];
+}
+
+- (void)logVisibleItems {
+  NSMutableArray<NSString *> *visibleItemTitles = [NSMutableArray array];
+  for (UITabBarItem *item in self.tabBar.items) {
+    CGRect itemViewInWindow = [self.tabBar rectForItem:item inCoordinateSpace:self.view.window];
+    CGRect overlapRect = CGRectIntersection(self.view.window.bounds, itemViewInWindow);
+    if (CGRectIsNull(overlapRect) || CGRectGetWidth(itemViewInWindow) < 1) {
```

Comment:
It's to avoid division by zero. This gets called before the subviews are laid-out during initial presentation. Since this was a hacky little demo, it doesn't have all the sharp edges worked-out yet.

Reply:
Could we rewrite it as `width > 0` to make it a bit more clear then?

Reply:
I originally wrote it as `< (CGFloat)0.001` but that seemed like it would generate questions about the specific choice of precision.  As long as this example doesn't crash, I don't have much opinion. I would expect that production code would be significantly more robust (and more efficient).

Reply:
Documented a bit more helpfully.